### PR TITLE
Add support up to python 3.11

### PR DESCRIPTION
**Pull request by kddejong**
*Issue #, if available:*

*Description of changes:*
- Add support up to python 3.11

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/rpdk/core/contract/resource_client.py [hunk 1]
```diff
@@ -458,8 +457,7 @@ def compare_model(self, inputs, outputs, path=()):
             "All properties specified in the request MUST "
             "be present in the model returned, and they MUST"
             " match exactly, with the exception of properties"
-            " defined as writeOnlyProperties in the resource schema \n Request Model : %s \n Returned Model : %s \n"
```
Comment (mircealam) [inline src/rpdk/core/contract/resource_client.py:461]: formatting and functional changes should probably be separate, but assuming this was done automatically so should be good

Comment (kddejong) [reply src/rpdk/core/contract/resource_client.py:461]: Yea some of this was the result of passing pylint updates.  I was debating separating it.

File: setup.py [hunk 2]
```diff
@@ -70,11 +70,10 @@ def find_version(*file_paths):
         "Topic :: Software Development :: Code Generators",
         "Operating System :: OS Independent",
         "Programming Language :: Python :: 3 :: Only",
-        "Programming Language :: Python :: 3.6",
-        "Programming Language :: Python :: 3.7",
         "Programming Language :: Python :: 3.8",
         "Programming Language :: Python :: 3.9",
         "Programming Language :: Python :: 3.10",
+        "Programming Language :: Python :: 3.11",
```
[the remark this answers is not in the document]
Comment (kddejong) [reply setup.py:76]: Lambda not support 3.12 makes for a weird angle.  The CLI could work in 3.12 but until Lambda supports it we can't develop a resource for it?

[the remark this answers is not in the document]
Comment (mircealam) [reply setup.py:76]: okay. we can defer adding 3.12

**Review by mircealam — APPROVED** [review]

**Review by mmaeng — APPROVED** [review]

**Pull request merged**